backend/tracim_backend/lib/webdav/dav_provider.py

```diff
@@ -1,33 +1,140 @@
 # coding: utf8
-from os.path import basename
-from os.path import dirname
+from collections import deque
+import enum
 import typing
 
 from pluggy import PluginManager
-from sqlalchemy.orm.exc import NoResultFound
 from wsgidav.dav_provider import DAVProvider
 from wsgidav.lock_manager import LockManager
 
-from tracim_backend.app_models.contents import content_type_list
 from tracim_backend.config import CFG
 from tracim_backend.exceptions import ContentNotFound
 from tracim_backend.exceptions import NotAuthenticated
 from tracim_backend.exceptions import WorkspaceNotFound
 from tracim_backend.lib.core.content import ContentApi
-from tracim_backend.lib.core.content import ContentRevisionRO
 from tracim_backend.lib.core.user import UserApi
 from tracim_backend.lib.core.workspace import WorkspaceApi
 from tracim_backend.lib.utils.request import TracimContext
 from tracim_backend.lib.utils.utils import normpath
 from tracim_backend.lib.utils.utils import webdav_convert_file_name_to_bdd
 from tracim_backend.lib.webdav import resources
 from tracim_backend.lib.webdav.lock_storage import LockStorage
+from tracim_backend.lib.webdav.resources import get_content_resource
+from tracim_backend.lib.webdav.resources import get_workspace_resource
+from tracim_backend.models.auth import User
 from tracim_backend.models.data import Content
-from tracim_backend.models.data import ContentNamespaces
 from tracim_backend.models.data import Workspace
 from tracim_backend.models.tracim_session import TracimSession
 
 
+class ResourceLevelType(enum.Enum):
+    ROOT = "root"
+    WORKSPACE = "workspace"
+    CONTENT = "content"
+
+
+class ProcessedWebdavPath(object):
+    """
+    Processor for Webdav Path:
+    - get all workspaces hierarchy from path
+    - get all content hierarchy from path
+    - provide useful properties to handle Webdav Request
```

Comment:
```suggestion
    - get the workspace hierarchy from path
    - get the content hierarchy from path
    - provide useful properties to handle the WebDAV request 
```
?

Reply:
i suggest : 
```suggestion
    - get the workspaces hierarchy from path
    - get the contents hierarchy from path
    - provide useful properties to handle Webdav Request
```
There is a hierarchy of multiple workspaces and an hierarchy of multiple contents.